File: CS/DXGrid_CustomGrouping/MyViewModel.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;

namespace DXGrid_CustomGrouping {
    public class MyViewModel  {
        public MyViewModel() {
            CreateList();
        }


        public ObservableCollection<Person> ListPerson { get; set; }
     
        void CreateList() {
            ListPerson = new ObservableCollection<Person>();
            for (int i = 0; i < 100; i++) {
                Person p = new Person(i);
                ListPerson.Add(p);
            }
        }
    }

    public class Person  {
        public Person() {

        }
        public Person(int i) {
            FirstName = "FirstName" + i;
            LastName = "LastName" + i;
          UnitPrice = i ;
        
        }

        string _firstName;

        public string FirstName {
            get { return _firstName; }
            set {
                _firstName = value;
             
            }
        }
        public string LastName { get; set; }
        int _age;

        public int UnitPrice {
            get { return _age; }
            set { _age = value; }
        }
     


    }
}

File: VB/DXGrid_CustomGrouping/MyViewModel.vb
Imports System.Collections.ObjectModel

Namespace DXGrid_CustomGrouping

    Public Class MyViewModel

        Public Sub New()
            CreateList()
        End Sub

        Public Property ListPerson As ObservableCollection(Of Person)

        Private Sub CreateList()
            ListPerson = New ObservableCollection(Of Person)()
            For i As Integer = 0 To 100 - 1
                Dim p As Person = New Person(i)
                ListPerson.Add(p)
            Next
        End Sub
    End Class

    Public Class Person

        Public Sub New()
        End Sub

        Public Sub New(ByVal i As Integer)
            FirstName = "FirstName" & i
            LastName = "LastName" & i
            UnitPrice = i
        End Sub

        Private _firstName As String

        Public Property FirstName As String
            Get
                Return _firstName
            End Get

            Set(ByVal value As String)
                _firstName = value
            End Set
        End Property

        Public Property LastName As String

        Private _age As Integer

        Public Property UnitPrice As Integer
            Get
                Return _age
            End Get

            Set(ByVal value As Integer)
                _age = value
            End Set
        End Property
    End Class
End Namespace
